### PR TITLE
[DOC] Add urldecode processor documentation

### PR DESCRIPTION
**Pull request by vagimeli**
### Description
Closes content gap

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/4647

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@hdhalter We need a dev to take the first pass at drafting this content. I can support them in refining the content and getting it through the documentation process. The ingest processor template is provided with this PR, so it should help the dev get started. Thanks!

File: _ingest-pipelines/processors/urldecode.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `urldecode` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field`  | Required  | The field containing the URL-encoded string to be decoded. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
```
Comment (vagimeli) [inline _ingest-pipelines/processors/urldecode.md:30]: @gaobinlong Are template snippets applicable for this processor's parameters?

Comment (gaobinlong) [reply _ingest-pipelines/processors/urldecode.md:30]: No, not applicable, both the `file` and `target_field` do not support template snippets.

File: _ingest-pipelines/processors/urldecode.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+`target_field`  | Required  | The field where the decoded string is stored. If not specified, the decoded string is stored in the same field as the original encoded string. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
+`ignore_missing`  | Optional  | Specifies whether the processor should ignore documents that do not contain the specified `field`. If set to `true`, the processor ignores missing values in the `field` and leaves the `target_field` unchanged. Default is `false`. |
+`override_target`  | Optional  | Determines what happens when `target_field` exists in the document. If set to `true`, the processor overwrites the existing `target_field` value with the new value. If set to `false`, the existing value remains and the processor does not overwrite it. Default is `false`. |
```
Comment (gaobinlong) [inline _ingest-pipelines/processors/urldecode.md:33]: This processor doesn't support `override_target`.

File: _ingest-pipelines/processors/urldecode.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `urldecode` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field`  | Required  | The field containing the URL-encoded string to be decoded. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
+`target_field`  | Required  | The field where the decoded string is stored. If not specified, the decoded string is stored in the same field as the original encoded string. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
```
Comment (gaobinlong) [inline _ingest-pipelines/processors/urldecode.md:31]: `target_field` is optional.

File: _ingest-pipelines/processors/urldecode.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `urldecode` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field`  | Required  | The field containing the URL-encoded string to be decoded. |
+`target_field`  | Optional  | The field where the decoded string is stored. If not specified, the decoded string is stored in the same field as the original encoded string. |
```
Comment (natebower) [inline _ingest-pipelines/processors/urldecode.md:31]:
```suggestion
`target_field`  | Optional  | The field in which the decoded string is stored. If not specified, then the decoded string is stored in the same field as the original encoded string. |
```

File: _ingest-pipelines/processors/urldecode.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+`ignore_missing`  | Optional  | Specifies whether the processor should ignore documents that do not contain the specified `field`. If set to `true`, the processor ignores missing values in the `field` and leaves the `target_field` unchanged. Default is `false`. |
```
Comment (natebower) [inline _ingest-pipelines/processors/urldecode.md:32]:
```suggestion
`ignore_missing`  | Optional  | Specifies whether the processor should ignore documents that do not contain the specified `field`. If set to `true`, then the processor ignores missing values in the `field` and leaves the `target_field` unchanged. Default is `false`. |
```

File: _ingest-pipelines/processors/urldecode.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+`description`  | Optional  | A brief description of the processor.  |
+`if` | Optional | A condition for running the processor. |
+`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters an error. If set to `true`, failures are ignored. Default is `false`. |
```
Comment (natebower) [inline _ingest-pipelines/processors/urldecode.md:35]:
```suggestion
`ignore_failure` | Optional | Specifies whether the processor continues to run even if it encounters an error. If set to `true`, then failures are ignored. Default is `false`. |
```

File: _ingest-pipelines/processors/urldecode.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+PUT testindex1/_doc/1?pipeline=url_decode_pipeline
+{
+  "encoded_url": "https://example.com/search?q=url%20decode%20test"
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The request indexes the document into the index `testindex1` and will index all documents with the
```
Comment (natebower) [inline _ingest-pipelines/processors/urldecode.md:122]: Should we be referring to the response in this text, or should the heading be "Request"? Should this text end in a colon?

Comment (natebower) [inline _ingest-pipelines/processors/urldecode.md:122]:
```suggestion
The request indexes the document into the index `testindex1` and will index all documents containing the 
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

**Review by natebower — APPROVED** [review]
LGTM

**Pull request merged**